cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceRewriter.java

```diff
@@ -255,29 +277,69 @@ public void visitEnd() {
             // method parameter with this.something
             if (name.startsWith("this.")) {
               String fieldName = getFieldName(name);
```

Comment:
No need to have the `getFieldName` method since you already checked name starts with `"this"`, simply do a `name.substring("this.".length());` would do.

Reply:
Also, isn't that you need to check if `fieldDetails` contains the field name as well?

Reply:
The whole logic could be simpler if you do:

```java
for (String name : nodeProcessor.getEntities()) {
  if (paramPositions.containsKey(name)) {   // Param name cannot have ".", so it won't have "this." if it matches
    entityPartDetails.add(new EntityPartDetail(name, false));
    continue;
  }
  name = name.startsWith("this.") ? name.substring("this.".length()) : name;
  if (!fieldDetails.containsKey(name)) {
    throw new IllegalArgumentException(....);
  }
  entityPartDetails.add(new EntityPartDetail(name, true));
}
```